build.sbt

```diff
@@ -62,14 +62,17 @@ lazy val sharedSettings = mimaDefaultSettings ++ Seq(
 
 lazy val js = project.in(file("js"))
   .settings(sharedSettings: _*)
-  .settings(scalaJSSettings: _*)
   .settings(
-    libraryDependencies += "org.scala-lang.modules.scalajs" %% "scalajs-test-bridge" % scalaJSVersion,
-    ScalaJSKeys.scalaJSTestFramework in Test := "org.scalacheck.ScalaCheckFramework"
-  )
+    scalaJSStage in Global := FastOptStage,
+    parallelExecution in Test := false,
+    fork := false,
+    libraryDependencies += "org.scala-js" %% "scalajs-test-interface" % scalaJSVersion,
+    testFrameworks := Seq(new TestFramework("org.scalacheck.ScalaCheckFramework"))
+  ).enablePlugins(ScalaJSPlugin)
 
 lazy val jvm = project.in(file("jvm"))
   .settings(sharedSettings: _*)
   .settings(
-    libraryDependencies += "org.scala-sbt" %  "test-interface" % "1.0"
-  )
+    libraryDependencies += "org.scala-sbt" %  "test-interface" % "1.0",
+    libraryDependencies += "org.scala-js" %% "scalajs-stubs" % scalaJSVersion
```

Comment:
You probably want to put this in `% "provided"` so that it does not pollute the publicly visible dependencies in the JVM artifact. Note that this project contains only macros in the JVM version.